tests/framework/db/ActiveRecordTest.php

```diff
@@ -1439,4 +1439,20 @@ public function testUnlinkAllOnConditionViaTable()
         // Make sure that only links were removed, the items were not removed
         $this->assertEquals(3, $itemClass::find()->where(['category_id' => 2])->count());
     }
+
+    /**
+     * https://github.com/yiisoft/yii2/pull/13891
+     */
+    public function testIndexByAfterLoadingRelations()
+    {
+        $orderClass = $this->getOrderClass();
+
+        $this->assertNotEmpty($orderClass::find()->with('customer')->indexBy(function($order) {
```

Comment:
why does the `with()` statement calls the customer relation and the index uses the product relation?